python/cudf/cudf/core/tools/datetimes.py

```diff
@@ -652,3 +653,26 @@ def _isin_datetimelike(
 
     res = lhs._obtain_isin_result(rhs)
     return res
+
+
+def to_iso_calendar(self):
+    formats = ["%G", "%V", "%u"]
+    if isinstance(
+        self,
+        (
+            cudf.core.series.DatetimeProperties,
+            pd.core.indexes.accessors.DatetimeProperties,
+        ),
+    ):
+        iso_params = [self.strftime(fmt) for fmt in formats]
+        index = None
```

Comment:
I believe this `index = None` (when passed to the DataFrame constructor) will throw away an existing index if it's not a RangeIndex. I believe the  `DatetimeProperties` class has access to the underlying series (at least for cuDF), which means we should hopefully be able to maintain it.

Reply:
Yes, that's correct, I was trying to allow for the `index` value that's needed below, but I don't need to set it to None. I'll change the code here!